src/discord-cluster-manager/leaderboard_db.py

```diff
@@ -163,22 +164,26 @@ def get_leaderboard(self, leaderboard_name: str) -> int | None:
     def get_leaderboard_submissions(
         self, leaderboard_name: str
     ) -> list[SubmissionItem]:
-        """
-        TODO: Change these all to be IDs instead
-        """
         self.cursor.execute(
-            "SELECT * FROM submissions WHERE leaderboard_id = %s ORDER BY submission_score ASC",
+            """
+            SELECT s.name, s.user_id, s.code, s.submission_time, s.score
+            FROM leaderboard.submission s
+            JOIN leaderboard.problem p
+            ON s.problem_id = p.id
+            WHERE p.name = %s
+            ORDER BY s.score ASC
+            """,
             (leaderboard_name,),
         )
 
         return [
             SubmissionItem(
-                leaderboard_name=submission[1],
-                submission_name=submission[2],
-                user_id=submission[3],
-                code=submission[4],
-                submission_time=submission[5],
-                submission_score=submission[6],
+                leaderboard_name=leaderboard_name,
```

Comment:
@b9r5 Could you look into a way to do this cleaner way? This was a very quick way to make it work, but would love to have a cleaner way to construct those models. This is very error prone.
Feel free to do that in a new PR, can merge this one.